<launch>
    <arg name="marker_size" doc="Size of the ArUco marker used, in meters" default="0.1" />
    <arg name="marker_id" doc="The ID of the ArUco marker used" default="100" />
    <arg name="use_rviz" default="true" />

    <!-- start the Realsense -->
    <include file="$(find realsense2_viewer)/launch/realsense2_viewer.launch" />

    <!-- start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" />
        <param name="image_is_rectified" value="true" />
        <param name="marker_size"        value="$(arg marker_size)" />
        <param name="marker_id"          value="$(arg marker_id)" />
        <param name="reference_frame"    value="camera_color_optical_frame" />
        <param name="camera_frame"       value="camera_color_optical_frame" />
        <param name="marker_frame"       value="aruco_marker" />
    </node>

    <node if="$(arg use_rviz)" name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false" 
        args="-d $(find mantra_application)/calibration/config/aruco_tracker.rviz" output="screen" />

</launch>